src/render/src/App.svelte

```diff
@@ -210,16 +282,50 @@ async function createCFG(params: Params): Promise<CFG> {
   }
 }
 
+let functionAndCFGMetadata: FunctionAndCFGMetadata | undefined;
+
+function updateMetadata(CFG: CFG, func?: SyntaxNode, language?: Language) {
+  // Update function metadata (GitHub only)
+  let name: string | undefined = undefined;
+  let lineCount: number | undefined = undefined;
+
+  if (func && language) {
+    name = extractFunctionName(func, language) ?? "<anonymous>";
+    lineCount = func.endPosition.row - func.startPosition.row + 1;
+  }
+
+  // Update CFG metadata
+  const nodeCount: number = CFG.graph.order;
+  const edgeCount: number = CFG.graph.size;
+  const cyclomaticComplexity: number = CFG.graph.size - nodeCount + 2; // (https://en.wikipedia.org/wiki/Cyclomatic_complexity)
+
+  return {
+    functionData: name && lineCount ? { name, lineCount, language } : undefined,
+    cfgGraphData: {
+      nodeCount,
+      edgeCount,
+      cyclomaticComplexity,
+    },
+  };
+}
+
 async function render() {
   try {
     const urlSearchParams = new URLSearchParams(window.location.search);
     const params = parseUrlSearchParams(urlSearchParams);
     setBackgroundColor(params.colors);
+
+    const cfg = await createCFG(params);
+
     if (params.type === "GitHub") {
       codeUrl = params.codeUrl;
+      const { func, language } = await fetchFunctionAndLanguage(params);
```

Comment:
I don't like that we're fetchign this twice, but I guess we'll have to do for now...